specs/data_structures.md

```diff
@@ -379,16 +380,21 @@ Finally, the `availableDataRoot` of the block [Header](#header) is computed as t
 
 A share is a fixed-size data chunk that will be erasure-coded and committed to in [Namespace Merkle trees](#namespace-merkle-tree).
 
-| name      | type               | description     |
-| --------- | ------------------ | --------------- |
-| `rawData` | `byte[SHARE_SIZE]` | Raw share data. |
+| name          | type                         | description                |
+| ------------- | ---------------------------- | -------------------------- |
+| `namespaceID` | [NamespaceID](#type-aliases) | Namespace ID of the share. |
```

Comment:
A LL message is associated with a namespaceID. What is the namespace of a share though? Can't a share contain the end of one message and the beginning of another one (with different NIDS)?

Reply:
No. From the rationale doc (https://github.com/lazyledger/lazyledger-specs/blob/adlerjohn-share_namespace/rationale/message_block_layout.md):

> 1. Data must be ordered by namespace ID. This makes queries into a NMT commitment of that data more efficient.
> 2. Since non-message data are not naturally intended for particular namespaces, we assign reserved namespaces for them. A range of namespaces is reserved for this purpose, starting from the lowest possible namespace ID.
> 3. By construction, the above two rules mean that non-message data always precedes message data in the row-major matrix, even when considering single rows or columns.
> 4. Data with different namespaces must not be in the same share. This might cause a small amount of wasted block space, but makes the NMT easier to reason about in general since leaves are guaranteed to belong to a single namespace.

By (4) shares will only have a single namespace ID associated with them. The reason shares have namespace IDs is that the NMT has shares as its leaves, and we need to know the NIDs of each leaf (i.e. share).

Reply:
To expand on that, the NMT doesn't care if it's hashing messages or transactions or evidence. It just sees shares. And each share needs to have a NID somehow.

Reply:
I see. And the order for account balance related Tx is preserved because they end up in the right order in their preserved namespace?